tests/testthat/test-rdiffnet.R

```diff
@@ -106,3 +106,59 @@ test_that("Simulation study", {
   expect_equal(ans0, ans1)
 
 })
+
+# Test for multi diffusion ---
+
+# Seed of first adopters
+test_that("All should be equal! (multiple)", {
+  set.seed(12131)
+  n            <- 50
+  t            <- 5
+  graph        <- rgraph_ws(n, 4, p=.3)
+  seed.p.adopt <- list(0.1, 0.1)
+  seed.nodes   <- c(1,5,7,10)
+  thr          <- runif(n, .2,.4)
+  thr_list     <- list(thr,thr)
+
+  # Generating identical networks
+  net1 <- rdiffnet(seed.graph = graph, seed.nodes = seed.nodes, seed.p.adopt = seed.p.adopt,
+                   t = t, rewire = FALSE, threshold.dist = thr_list)
+
+  net2 <- rdiffnet(seed.graph = graph, seed.nodes = seed.nodes, seed.p.adopt = seed.p.adopt,
+                   t = t, rewire = FALSE, threshold.dist = thr_list)
+
+  expect_equal(net1, net2)
+})
+
+
+#single
+rdiffnet(100, 5)
+rdiffnet(100, 5, seed.p.adopt = 0.1)
+rdiffnet(100, 5, seed.p.adopt = 0.1, seed.nodes = 'random')
+rdiffnet(100, 5, seed.nodes = c(1,3,5))
+net_1 <- rdiffnet(100, 5, seed.nodes = c(1,3,5))
+summary(net_1)
+
+#multi
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), behavior = c('tabacco', 'alcohol'))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), seed.nodes = 'random')
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), seed.nodes = c('random', 'central'))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = 0.3)
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = list(0.1,0.2))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = rexp(100))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = list(rexp(100),runif(100)))
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = function(x) 0.3)
+rdiffnet(100, 5, seed.p.adopt = list(0.1,0.08), threshold.dist = list(function(x) 0.3, function(x) 0.2))
```

Comment:
Perhaps, add expectations of of the sort of "expect class" something like that so you ensure it returns diffnet objects.